dune/dune.go

```diff
@@ -161,3 +165,21 @@ func (c *duneClient) QueryResults(executionID string) (*models.ResultsResponse,
 
 	return &resultsResp, nil
 }
+
+func (c *duneClient) QueryResultsCSV(executionID string) (io.Reader, error) {
+	resultsURL := fmt.Sprintf(resultsCSVURLTemplate, c.env.Host, executionID)
+	req, err := http.NewRequest("GET", resultsURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := httpRequest(c.env.APIKey, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// we read whole result into ram here. if there was a paginated API we wouldn't need to
```

Comment:
I don't think you need to tho. You can set up a ring buffer that reads from the http response when there's new stuff and space in the ring buffer and allows the user to read from it using the standard io.Reader interface.

I have used [this library](https://github.com/glycerine/rbuf) for my streaming PoC I wrote a while ago and it worked quite well. It's pretty simple, does what it says and not more.

I'd rather we keep our library low on memory requirements and only use a lot of memory when absolutely necessary